packages/chain-adapters/src/cosmossdk/osmosis/OsmosisChainAdapter.ts

```diff
@@ -0,0 +1,73 @@
+import { bip32ToAddressNList, OsmosisSignTx, OsmosisWallet } from '@shapeshiftoss/hdwallet-core'
+import { BIP44Params, chainAdapters, ChainTypes } from '@shapeshiftoss/types'
+
+import { ChainAdapter as IChainAdapter } from '../../api'
+import { ErrorHandler } from '../../error/ErrorHandler'
+import { toPath } from '../../utils'
+import { ChainAdapterArgs, CosmosSdkBaseAdapter } from '../CosmosSdkBaseAdapter'
+export class ChainAdapter
+  extends CosmosSdkBaseAdapter<ChainTypes.Osmosis>
+  implements IChainAdapter<ChainTypes.Osmosis>
+{
+  protected readonly supportedChainIds = ['cosmos:osmosis-1', 'cosmos:osmo-testnet-1']
+  protected readonly chainId = this.supportedChainIds[0]
+
+  public static readonly defaultBIP44Params: BIP44Params = {
+    purpose: 44,
+    coinType: 118,
+    accountNumber: 0
+  }
+
+  constructor(args: ChainAdapterArgs) {
+    super(args)
+  }
+
+  getType(): ChainTypes.Osmosis {
+    return ChainTypes.Osmosis
+  }
+
+  async getAddress(input: chainAdapters.GetAddressInput): Promise<string> {
+    const { wallet, bip44Params = ChainAdapter.defaultBIP44Params } = input
+    const path = toPath(bip44Params)
+    const addressNList = bip32ToAddressNList(path)
+    const osmosisAddress = await (wallet as OsmosisWallet).osmosisGetAddress({
+      addressNList,
+      showDisplay: Boolean(input.showOnDevice)
+    })
+    return osmosisAddress as string
```

Comment:
why the cast? i'm assuming `osmosisGetAddress` can return `string | undefined`?

Reply:
would be the same fix as related to this comment: https://github.com/shapeshift/lib/pull/397#discussion_r814962963 in commit: [c631793](https://github.com/shapeshift/lib/pull/397/commits/c631793abc28fbe0770d544fd394ec035e626905)